dib/generate_dag.go

```diff
@@ -114,7 +114,7 @@ func generateHashes(graph *dag.DAG, allFiles []string) error {
 	graph.WalkInDepth(func(node *dag.Node) {
 		nodeFiles[node] = []string{}
 		for _, file := range allFiles {
-			if !strings.HasPrefix(file, node.Image.Dockerfile.ContextPath) {
+			if path.Dir(file) != node.Image.Dockerfile.ContextPath {
```

Comment:
what about files that belong to subdirectories ?

For instance
```
root
--- Dockerfile
--- sub-dir1
      --- surb-dir2
            --- somefile
```

Reply:
Indeed. I did this to avoid conflicts when two directories have a similar name :
```
├── php/
│   └── Dockerfile
└── php-nodejs/
    └── Dockerfile
```
I'll find a better solution